File: week_10/c_sharp_course/week_1/day_4/composition/checkpoint_1_composition/wizard_management_specs/WizardTest.cs
using NUnit.Framework;

namespace WizardManagement { 

  [TestFixture]
  public class WizardTest {
    Wizard wizard;
    Broomstick broomstick;

    [SetUp]
    public void Init() 
    {
      broomstick = new Broomstick("Nimbus", 10);
      wizard = new Wizard("Toby", broomstick);
    }

    [Test]
    public void HasName()
    {
      Assert.AreEqual("Toby", wizard.Name);
    }

    [Test]
    public void HasBroomstick()
    {
      Broomstick ride = (Broomstick) wizard.Ride;
      Assert.AreEqual("Nimbus", ride.Brand);
    }

    [Test]
    public void CanFly()
    {
      Assert.AreEqual("mounting broom, running, skipping, flying!", wizard.Fly());
    }

    [Test]
    public void CanFlyDragon(){
      Dragon dragon = new Dragon("Tilly");
      wizard = new Wizard("Toby", dragon);
      Assert.AreEqual("Standing up tall, beating wings, lift off!", wizard.Fly());
    }

    [Test]
    public void CanFlyMagicCarpet(){
      MagicCarpet carpet = new MagicCarpet("Purple");
      wizard = new Wizard("Toby", carpet);
      Assert.AreEqual("Hovering up, straightening out, flying off!", wizard.Fly());
    }

    [Test]
    public void CanSetRide(){
      Dragon dragon = new Dragon("Erik");
      wizard.Ride = dragon;
      Assert.AreEqual("Standing up tall, beating wings, lift off!",wizard.Fly());
    }
  }
}

File: week_06/day_3/composition/Composition_Lesson_End/app/src/test/java/example/codeclan/com/composition_lesson_end/WizardTest.java
package example.codeclan.com.composition_lesson_end;

/**
 * Created by user on 28/08/2017.
 */

import static org.junit.Assert.*;
import org.junit.*;

import example.codeclan.com.composition_lesson_end.wizard_management.*;


public class WizardTest {

    Wizard wizard;
    Broomstick broomstick;

    @Before
    public void before(){
        broomstick = new Broomstick("Nimbus", 10);
        wizard = new Wizard("Toby", broomstick);
    }

    @Test
    public void hasName(){
        assertEquals("Toby", wizard.getName());
    }

    @Test
    public void hasRide(){
        assertNotNull(wizard.getRide());
    }

    @Test
    public void canSetRide(){
        Dragon dragon = new Dragon("Erik");
        wizard.setRide(dragon);
        assertEquals("Standing up tall, beating wings, lift off!", wizard.fly());
    }

    @Test
    public void canFlyBroomstick(){
        assertEquals("mounting broom, running, skipping, flying!", wizard.fly());
    }

    @Test
    public void canFlyDragon(){
        Dragon dragon = new Dragon("Tilly");
        wizard = new Wizard("Toby", dragon);
        assertEquals("Standing up tall, beating wings, lift off!", wizard.fly());
    }

    @Test
    public void canFlyMagicCarpet(){
        MagicCarpet carpet = new MagicCarpet("Purple");
        wizard = new Wizard("Toby", carpet);
        assertEquals("Hovering up, straightening out, flying off!", wizard.fly());
    }
}
